### PR TITLE
[FLINK-11293][kafka, test] Added timeout to delete topic request.

### PR DESCRIPTION
**Pull request by dawidwys**
## What is the purpose of the change

The purpose of this PR is to harden kafka ITCases. It targets probable kafka AdminClient connection issues.

## Brief change log

Logs from failures show that even though the request never finishes,
kafka topics are actually deleted. Therefore this PR adds timeout on
delete request. If the request times out we check if the topic actually
exists. If not then we continue as if the delete request succeeded.

## Verifying this change

Current kafka ITCases still pass.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no /** don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by dawidwys** [comment]
@pnowojski Could you have a look at this PR?

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTestEnvironmentImpl.java [hunk 1]
```diff
@@ -155,10 +159,23 @@ public void prepare(Config config) {
...
+
+	private void tryDelete(AdminClient adminClient, String topic)
+			throws ExecutionException, InterruptedException, TimeoutException {
+		try {
+			adminClient.deleteTopics(Collections.singleton(topic)).all().get(DELETE_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+		} catch (TimeoutException e) {
+			LOG.info("Did not receive delete topic response within %d seconds. Checking if it succeeded",
+				DELETE_TIMEOUT_SECONDS);
+			if (adminClient.listTopics().names().get(DELETE_TIMEOUT_SECONDS, TimeUnit.SECONDS).contains(topic)) {
+				fail(String.format("Delete test topic : %s failed.", topic));
```
Comment (pnowojski) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTestEnvironmentImpl.java]: replace `fail(...)` with `throw e;`? Would deduplicate the error handling?

**Comment by dawidwys** [comment]
I don't think we need to port it to previous versions. As far as I understand the code previous versions do it in a totally different way, they connect to zookeeper directly. All the failures reported so far were from the kafka 2.0.

**Comment by dawidwys** [comment]
Hi @pnowojski 
I've looped tests on travis. I run tests (all tests in `flink-connector-kafka`) with that change and without:
- with the fix - 100 times -> 0 failures related to deleting topics
- without the fix - 100 times -> 3 failures related to deleting topics
Do you think it is a sufficient proof that this fix improves the stability?

**Comment by dawidwys** [comment]
Hi @pnowojski

I've looped tests on travis. I run tests (all tests in `flink-connector-kafka`) with that change and without:
- with the fix - 100 times -> 0 failures related to deleting topics
- without the fix - 100 times -> 3 failures related to deleting topics

Do you think it is a sufficient proof that this fix improves the stability?

**Pull request merged**